sdk/src/main/kotlin/com/pulumi/kotlin/Common.kt

```diff
@@ -31,7 +31,7 @@ interface ConvertibleToJava<T> {
  * it cannot be sealed, because generated subclasses will be placed in other packages.
  */
 @Suppress("UnnecessaryAbstractClass")
-abstract class KotlinResource private constructor(internal open val javaResource: JavaResource) {
+abstract class KotlinResource private constructor(protected open val javaResource: JavaResource) {
```

Comment:
Does this (`internal` -> `protected`) solve some technical problem or are we trying to hide it from the user here?

Reply:
I'm asking because we have `protected` `javaResource` and `internal` `underLyingJavaResource`. User still has access to `underLyingJavaResource`.

Woulnd't having just a single `internal` `javaResource` field be the same here?

Reply:
Keeping the resource as `internal` caused issues in subclasses (since they're not in this module anymore):
<img width="1383" alt="image" src="https://github.com/VirtuslabRnD/pulumi-kotlin/assets/34098778/f4fe10b0-c124-4077-b0d2-29d29d4834e5">

Keeping the resource as `protected` caused issues in the `Alias` class (which is inside the same module, but it's not a subclass).

> User still has access to `underLyingJavaResource`.

I thought it wouldn't be available to users of the published library?

Reply:
Ok, I understand now 👍


> > User still has access to `underLyingJavaResource`.
> 
> I thought it wouldn't be available to users of the published library?

Yes, you are right.

Reply:
Ok, I understand now 👍


> > User still has access to `underLyingJavaResource`.
> 
> I thought it wouldn't be available to users of the published library?

Yes, you are right.